### ISSUE TITLE
DAC build is not incremental

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#9932]
There are two rules related to the DAC build that cause our dotnet/coreclr builds to not be incremental (on Windows):

src/dlls/mscordac/CMakeLists.txt:

This command is always executed:
```
    # Generate export file
    add_custom_command(TARGET mscordaccore
        PRE_LINK
        COMMAND lib.exe /OUT:"${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/mscordaccore.lib" /DEF:"${CMAKE_CURRENT_BINARY_DIR}/mscordac.def" "$<TARGET_FILE:daccess>" $<$<OR:$<CONFIG:Release>,$<CONFIG:Relwithdebinfo>>:/LTCG> ${STATIC_LIBRARY_FLAGS} "${CMAKE_CURRENT_BINARY_DIR}/mscordaccore.dir/$<CONFIG>/mscordac.obj"
        COMMENT "Generating mscordaccore.exp export file"
    )
```

src/dlls/mscoree/coreclr/CMakeLists.txt:

This command is always executed:
```
    add_custom_command(
        TARGET coreclr
        POST_BUILD
        COMMAND ${CMAKE_CXX_COMPILER} /P /EP /TP ${PREPROCESS_DEFINITIONS} ${INC_DIR} /Fi${CMAKE_CURRENT_BINARY_DIR}/daccess.i  ${CLR_DIR}/src/debug/daccess/daccess.cpp
        COMMAND ${BuildToolsDir}/dactablegen.exe /dac:${CMAKE_CURRENT_BINARY_DIR}/daccess.i /pdb:${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/coreclr.pdb /dll:$<TARGET_FILE:coreclr> /bin:${CMAKE_CURRENT_BINARY_DIR}/wks.bin
        COMMAND ${BuildToolsDir}/InjectResource.exe /bin:${CMAKE_CURRENT_BINARY_DIR}/wks.bin /dll:$<TARGET_FILE:coreclr>
        COMMAND ${BuildToolsDir}/GenClrDebugResource.exe /dac:$<TARGET_FILE:mscordaccore> /dbi:$<TARGET_FILE:mscordbi> /sku:onecoreclr /out:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin
        COMMAND ${BuildToolsDir}/InjectResource.exe /bin:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin /dll:$<TARGET_FILE:coreclr> /name:CLRDEBUGINFO
        COMMENT Add dactable & debug resources to coreclr
    )
```

**Comment by BruceForstall** [comment]
cc @janvorli @noahfalk @jkotas

**Comment by BruceForstall** [comment]
I tried many variations of converting these to something of the form:
```
add_custom_command(OUTPUT foo DEPENDS ... COMMAND ...)
add_custom_target(foo_target DEPENDS foo)
add_dependencies(bar foo_target)
```
but could never get any of them to work.

For the "inject dac to coreclr" it also seems like we need to create a "dac_was_injected_sentinel.txt" file after the injection, with a timestamp of when that happened, so the rule can give the sentinel as OUTPUT and coreclr.dll as the DEPENDS.

**Comment by janvorli** [comment]
@BruceForstall I have managed to fix the first rule, I'll send out a PR on Monday.

**Comment by BruceForstall** [comment]
This issue should be fixed, with #10045
